### PR TITLE
Allow option :do in form_group to override input builder

### PR DESCRIPTION
**Pull request by nardopessoa**

### CODE REVIEW

File: lib/ex_bs/form.ex [hunk 1]
```diff
@@ -394,19 +419,22 @@ defmodule ExBs.Form do
...
 
       siblings ->
-        block = [input(form, field, opts) | errors(form, field)]
-        Map.put(data, :safe, [input_group(siblings, do: block) | safe])
+        Map.put(data, :safe, [input_group(siblings, do: content) | safe])
     end
   end
 
-  defp input_state_class(form, field) do
+  def input_state_class(form, field) do
```
Comment (abitdodgy) [inline lib/ex_bs/form.ex]: Do we want this to be public?

File: test/ex_bs/form_test.exs [hunk 2]
```diff
@@ -105,6 +105,37 @@ defmodule ExBs.FormTest do
       assert safe_to_string(input) == expected
     end
 
+    test "accepts custom input component", %{form: form} do
+      expected = fn form_group_class ->
```
[the remark this answers is not in the document]
Comment (nardopessoa) [reply test/ex_bs/form_test.exs]: I reuse the expected values in two asserts below:
1 - for `Form.form_group/3` with `do` block and without `options` arg
2 - for `Form.form_group/4` with `do` block and `options` arg

The difference between the expected results is the `class` passed on `options` arg. Do you think is better to create two separate test scenarios?

File: test/ex_bs/form_test.exs [hunk 3]
```diff
@@ -105,6 +105,37 @@ defmodule ExBs.FormTest do
...
+        ~s(<div class=\"form-group#{form_group_class}\">) <>
+          ~s(<label for=\"user_age\">Age</label>) <>
+          ~s(<select class="custom-select" id="user_age" name="user[age]">) <>
+          ~s(<option value="16">16</option>) <>
+          ~s(<option value="17">17</option>) <>
+          ~s(<option value="18">18</option>) <>
+          ~s(<option value="19">19</option>) <>
+          ~s(<option value="20">20</option>) <>
```
[the remark this answers is not in the document]
Comment (nardopessoa) [reply test/ex_bs/form_test.exs]: I agree. At this time, I thought of a real age selection scenario heheh sorry about that!

File: lib/ex_bs/form.ex (hunk 1 above)
Comment (nardopessoa) [reply lib/ex_bs/form.ex]: When I use `do block` for `Form.form_group`, the overridden input must receive the style for validated form, so I needed to use this function directly in the template.

Maybe, we must add this usage in `form_group/4` documentation example

**Pull request merged**